### ISSUE TITLE
Unison volume permissions are incorrect

### ISSUE THREAD

**Issue by achekulaev** [docksal/docksal#462]
Unison volume sets all initial permissions to root/root instead of using UID and GID of the host user.

**Comment by shelane** [comment]
This issue occurs on every sync from host to container... I switch branches in git and my host files are updated. Files are synced to container and again belong to root resulting in errors.

**Comment by achekulaev** [comment]
Yeah, but it seems to be non-functional. When I try to use those values, `bg-sync` container just crashes and fin left hanging, eternally waiting for Unison to finish it's sync...

**docksal.yml**

```
services:
  # Add environment vars
  bg-sync:
    environment:
      - UNISON_USER=docker
      # id -u
      - UNISON_UID=502
      - UNISON_GROUP=dialout
      # id -g
      - UNISON_GID=20
```

**docksal.env**

```
DOCKSAL_VOLUMES=unison
```

`fin project start` keeps `Waiting for Unison response...`

`bg-sync` has exited

```
$ fin ps
demo_bg-sync_1   bg-sync                       Exit 1      
...
```

Nothing useful in logs.

```
:: fin docker logs demo_bg-sync_1
addgroup: group 'dialout' in use

==> Setting up non-root user docker.

```

It crashed while tried to set up non-root user (here https://github.com/cweagans/docker-bg-sync/blob/master/sync.sh#L22)
Basically we need to fix this image first to use it. 
Hence the delay.

**Comment by shelane** [comment]
Yeah, I tried and had the same results. There is a pull request in for bg-sync with a hopeful fix. However, the maintainer asked the submitter to change how it was doing the pull request because it basically looked like everything changed. I pulled down both versions and did a diff with a different tool to see what he did. I thought of doing a pull request with just the changes to the user setup:
```
 # Create non-root user
   if [[ "$UNISON_USER" != "root" ]]; then
     log_heading "Setting up non-root user ${UNISON_USER}."
     local HOME="/home/${UNISON_USER}"
 
     # create group only, if it not exists
     if ! grep -q "$UNISON_GROUP" /etc/group; then
       log_info "Created group"
       addgroup -g "$UNISON_GID" -S "$UNISON_GROUP"
     fi
 
     # create user only, if it not exists
     if ! grep -q "$UNISON_USER" /etc/passwd; then
       log_info "Created user"
       adduser -u "$UNISON_UID" -D -S -G "$UNISON_GROUP" "$UNISON_USER"
     fi
 
     # create unison directory
     mkdir -p "${HOME}/.unison" || true
 
     # own the home directory
     chown -R "${UNISON_USER}:${UNISON_GROUP}" "${HOME}"
   fi
```

I wish I knew enough to try testing this before submitting it though. It’s been sitting there for a couple of months.

**Comment by achekulaev** [comment]
@shelane I have initiated https://github.com/docksal/unison (not available on Docker Hub yet) where I have included that user/groups patch from cweagans repo and did some cleanup. I want to see if I will be able to make that image to actually work fine for Docksal needs.

**Comment by shelane** [comment]
I took a look at what you’re doing with that. Is there a way that I could test that on my local?

**Comment by shelane** [comment]
Can you help me understand with these two lines:
```
COPY sync.sh /usr/local/bin/bg-sync
RUN chmod +x /usr/local/bin/bg-sync
```
where the sync.sh file is at the time the Dockerimage is built locally? I know it's in the git repo, but I don't see where that gets to where it's going to be doing this copy.

**Comment by achekulaev** [comment]
You want to grab the latest develop version of fin (1.54.1).

Put `DOCKSAL_VOLUMES=unison` to `docksal.env`

And add this to `docksal.yml` (change UID, GID if you need to):

```
version: "2.1"

services:
  # Add environment vars
  bg-sync:
    image: docksal/unison
    environment:
      - UNISON_USER=docker
      - UNISON_UID=502
      - UNISON_GROUP=dialout
      - UNISON_GID=20
```

And it is working for me. BTW now you can track what it is doing in logs `fin logs bg-sync` in another tab.

**Comment by achekulaev** [comment]
I pushed another version of fin. 1.54.2. 
To test this one you will also need to grab latest stacks for I have updated `volumes-unison`.

With this fin and this volumes-unison all you need to do is to use `DOCKSAL_VOLUMES=unison` and it will append your current UID and GID automatically.
